src/main/java/com/thebois/models/world/World.java

```diff
@@ -3,13 +3,18 @@
 import java.util.ArrayList;
 
 import com.thebois.models.IFinder;
+import com.thebois.models.Position;
+import com.thebois.models.world.structures.IStructure;
+import com.thebois.models.world.structures.Structure;
+import com.thebois.models.world.structures.StructureType;
 
 /**
  * World creates a matrix and keeps track of all the structures and resources in the game world.
  */
 public class World implements IFinder {
 
     private ITerrain[][] terrainMatrix;
+    private ArrayList<IStructure> structures = new ArrayList<>();
```

Comment:
Think we sould stay consisten how we represent the world in the model. We will also use the matrix for pathfinding, so in the world we should have a `structureMatrix` and in `.getStructures` we convert it to an ArrayList (and deepcopy the elements) and return that.

Reply:
I thought about it when writing the code, but it felt wrong to have a matrix that would be mostly empty. We can discuss this tomorrow during the meeting.

Reply:
> Yeah, the pathfinding will need to be able to easily find neighbouring structures as well as terrain tiles. Same goes for resources whenever we implement them. They should all be matrices like `terrainMatrix` and returned in the same way that `getStructures` does.

Have I understood this corectly by saying: You want the structures to be stored in a Matrix like the tiles, but in the getter it should be return as a Arraylist(like it is now, but a deep copy instead)?

Reply:
Yes, exactly. Should be the same as for `ITerrain`.